src/Symfony/Component/Console/Attribute/Argument.php

```diff
@@ -66,27 +68,39 @@ public static function tryFrom(\ReflectionParameter $parameter): ?self
             throw new LogicException(\sprintf('The parameter "$%s" of "%s()" must have a named type. Untyped, Union or Intersection types are not supported for command arguments.', $name, $self->function));
         }
 
-        $parameterTypeName = $type->getName();
+        $self->typeName = $type->getName();
+        $isBackedEnum = is_subclass_of($self->typeName, \BackedEnum::class);
 
-        if (!\in_array($parameterTypeName, self::ALLOWED_TYPES, true)) {
-            throw new LogicException(\sprintf('The type "%s" on parameter "$%s" of "%s()" is not supported as a command argument. Only "%s" types are allowed.', $parameterTypeName, $name, $self->function, implode('", "', self::ALLOWED_TYPES)));
+        if (!\in_array($self->typeName, self::ALLOWED_TYPES, true) && !$isBackedEnum) {
+            throw new LogicException(\sprintf('The type "%s" on parameter "$%s" of "%s()" is not supported as a command argument. Only "%s" types and BackedEnum are allowed.', $self->typeName, $name, $self->function, implode('", "', self::ALLOWED_TYPES)));
         }
 
         if (!$self->name) {
             $self->name = (new UnicodeString($name))->kebab();
         }
 
-        $self->default = $parameter->isDefaultValueAvailable() ? $parameter->getDefaultValue() : null;
+        if ($parameter->isDefaultValueAvailable()) {
+            $self->default = $parameter->getDefaultValue() instanceof \BackedEnum
+                ? $parameter->getDefaultValue()->value
+                : $parameter->getDefaultValue();
+        }
 
         $self->mode = $parameter->isDefaultValueAvailable() || $parameter->allowsNull() ? InputArgument::OPTIONAL : InputArgument::REQUIRED;
-        if ('array' === $parameterTypeName) {
+        if ('array' === $self->typeName) {
             $self->mode |= InputArgument::IS_ARRAY;
         }
 
         if (\is_array($self->suggestedValues) && !\is_callable($self->suggestedValues) && 2 === \count($self->suggestedValues) && ($instance = $parameter->getDeclaringFunction()->getClosureThis()) && $instance::class === $self->suggestedValues[0] && \is_callable([$instance, $self->suggestedValues[1]])) {
             $self->suggestedValues = [$instance, $self->suggestedValues[1]];
         }
 
+        if ($isBackedEnum && ! $self->suggestedValues) {
```

Comment:
```suggestion
        if ($isBackedEnum && !$self->suggestedValues) {
```

Reply:
I'm getting used to the Doctrine CS.

Reply:
I know and this comment from me should normally be enforced by the cs fixer config by fabbot 

cc @fabpot